### PR TITLE
Branch attendance functionality

### PR DESCRIPTION
**Pull request by ldinghan**

### CODE REVIEW

File: src/main/java/seedu/address/model/attendance/AttendancePresence.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    /**
+     * Returns true if the given string is a valid presence, which can be either 'present' or 'absent'.
+     *
+     * @param test The string to check for validity.
+     * @return True if the string is a valid presence, false otherwise.
+     */
+    public static boolean isValidInput(String test) {
+        return test.equals("present") || test.equals("absent");
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/attendance/AttendancePresence.java]: Might want to use regex for this.
`public static boolean isValidInput(String test) {
    return test.matches("(?i)^(present|absent)$");
}`

^ not sure if correct haha :P

File: src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+public class TakeAttendanceCommandTest {
+
+    @Test
+    public void execute_takeAttendanceWithSingleStudent_success() throws CommandException {
+        Model model = new ModelManager();
+        Person student = new PersonBuilder().withName("Alice").build();
+        Person newStudent = new PersonBuilder().withName("Bob").build();
+        model.addPerson(student);
+        model.addPerson(newStudent);
+        Session currentSession = new Session(new SessionNumber("2"), student);
```
Comment (raydenlim) [inline src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java:32]: Not related to this file, but you might want to add a testcase for "floats" in SessionNumberTest.java.
I think it wasnt covered

File: src/test/java/seedu/address/model/attendance/AttendancePresenceTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
+package seedu.address.model.attendance;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+
+public class AttendancePresenceTest {
```
Comment (raydenlim) [inline src/test/java/seedu/address/model/attendance/AttendancePresenceTest.java:7]: might want to test for white spaces after present to cover the trim() part in your method
" String validPresence= "present     ";

**Review by raydenlim — COMMENTED** [review]
LGTM!

**Review by raydenlim — APPROVED** [review]
:)

File: src/main/java/seedu/address/model/attendance/AttendancePresence.java (hunk 1 above)
Comment (ldinghan) [reply src/main/java/seedu/address/model/attendance/AttendancePresence.java]: you are right

File: src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java (hunk 2 above)
Comment (ldinghan) [reply src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java:32]: `public static final String VALIDATION_REGEX = "^[0-9]\\d*$";`
this line in SessionNumber:14 ensures that only numbers are allows (no decimal point), hence check for "floats" may not be necessary

File: src/test/java/seedu/address/model/attendance/AttendancePresenceTest.java (hunk 3 above)
Comment (ldinghan) [reply src/test/java/seedu/address/model/attendance/AttendancePresenceTest.java:7]: Sounds good! Will add that, thanks!

**Comment by craigtonlian** [comment]
LGTM 🚀

File: src/test/java/seedu/address/logic/parser/TakeAttendanceCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.TakeAttendanceCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.attendance.AttendancePresence;
+import seedu.address.model.person.Name;
+import seedu.address.model.session.SessionNumber;
+
+public class TakeAttendanceCommandParserTest {
```
Comment (Originalidk) [inline src/test/java/seedu/address/logic/parser/TakeAttendanceCommandParserTest.java:14]: Just a nit, but perhaps you could also test for invalid inputs for the sessions, names or attendance presence?

**Review by Originalidk — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/attendance/AttendancePresence.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Represents the presence of a student in an attendance record.
+ * It can either be 'present' or 'absent'.
+ */
+public class AttendancePresence {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Presence field should only contain either 'present' or 'absent', and it should not be blank";
+    public static final String VALID_REGEX = "(?i)^(present|absent)$";
```
Comment (wesho1107) [inline src/main/java/seedu/address/model/attendance/AttendancePresence.java]: Not sure if it will be better, but have you considered using enum?

**Review by wesho1107 — APPROVED** [review]
Other than suggestions, LGTM!

File: src/main/java/seedu/address/model/attendance/AttendancePresence.java (hunk 5 above)
Comment (ldinghan) [reply src/main/java/seedu/address/model/attendance/AttendancePresence.java]: Great idea! Refactored AttendancePresence to use enum

**Pull request merged**